flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -0,0 +1,347 @@
+    }
+}
+
+SqlNode SqlCreateTable() :
+{
+    final SqlParserPos startPos;
+    SqlIdentifier tableName;
+    String tableType = null;
+    SqlNodeList primaryKeyList = null;
+    List<SqlNodeList> uniqueKeysList = null;
+    SqlNodeList columnList = SqlNodeList.EMPTY;
+    SqlWatermark watermark = null;
+	SqlCharStringLiteral comment = null;
+
+    SqlNodeList propertyList = null;
+    SqlNodeList partitionColumns = null;
+
+    SqlParserPos pos;
+}
+{
+    <CREATE> { startPos = getPos(); }
+
+    <TABLE>
+
+    tableName = CompoundIdentifier()
+    [
+        <LPAREN> { pos = getPos(); TableCreationContext ctx = new TableCreationContext();}
+        TableColumn(ctx)
+        (
+            <COMMA> TableColumn(ctx)
+        )*
+        {
+            pos = pos.plus(getPos());
+            columnList = new SqlNodeList(ctx.columnList, pos);
+            primaryKeyList = ctx.primaryKeyList;
+            uniqueKeysList = ctx.uniqueKeysList;
+            watermark = ctx.watermark;
+            tableType = ctx.tableType;
+        }
+        <RPAREN>
+    ]
+    [ <COMMENT> <QUOTED_STRING> {
+        String p = SqlParserUtil.parseString(token.image);
+        comment = SqlLiteral.createCharString(p, getPos());
+    }]
+    [
+        <PARTITIONED> <BY>
+            {
+                SqlNode column;
+                List<SqlNode> partitionKey = new ArrayList<SqlNode>();
+                pos = getPos();
+
+            }
+            <LPAREN>
+            [
+                column = SimpleIdentifier()
+                {
+                    partitionKey.add(column);
+                }
+                (
+                    <COMMA> column = SimpleIdentifier()
+                        {
+                            partitionKey.add(column);
+                        }
+                )*
+            ]
+            <RPAREN>
+            {
+                partitionColumns = new SqlNodeList(partitionKey, pos.plus(getPos()));
+            }
+    ]
+    [
+        <WITH>
+            {
+                SqlNode property;
+                List<SqlNode> proList = new ArrayList<SqlNode>();
+                pos = getPos();
+            }
+            <LPAREN>
+            [
+                property = PropertyValue()
+                {
+                proList.add(property);
+                }
+                (
+                <COMMA> property = PropertyValue()
+                    {
+                    proList.add(property);
+                    }
+                )*
+            ]
+            <RPAREN>
+        {  propertyList = new SqlNodeList(proList, pos.plus(getPos())); }
+    ]
+
+    {
+        return new SqlCreateTable(startPos.plus(getPos()),
+                tableType,
+                tableName,
+                columnList,
+                primaryKeyList,
+                uniqueKeysList,
+                watermark,
+                propertyList,
+                partitionColumns,
+                comment);
+    }
+}
+
+SqlNode SqlDropTable() :
```

Comment:
Should we support this in another PR? Because the JIRA issue title only mentions create table DDL.

Reply:
The drop table grammar is so small for a PR merge, let's put them together and i will change the title